pkg/operator/mon/health.go

```diff
@@ -161,7 +161,7 @@ func (c *Cluster) checkMonsOnSameNode() (bool, error) {
 	nodesUsed := map[string]struct{}{}
 	for name, node := range c.mapping.Node {
 		// when the node is already in the list we have more than one mon on that node
-		if _, ok := nodesUsed[node.Name]; ok {
+		if _, ok := nodesUsed[node.Address]; ok {
```

Comment:
Instead of the `Name`, we will need to use the Label that has the name `kubernetes.io/hostname`. If we don't find that label then we could fall back to the name. So wherever we need the node name we would probably want to call a helper method that does this.

Reply:
Fully agree! Something that receives the `node` objects and returns `Address` if defined, and `Name` otherwise.